antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -16,7 +25,189 @@
+                matrix_target_paths=[matrix_node_lt.infer_target_path(is_link_lt)],
+            )
+        elif new_operator == BindingConstraintOperator.GREATER:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_gt.infer_target_path(is_link_gt)],
+            )
+        else:
+            raise NotImplementedError(
+                f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+            )
+    else:
+        raise NotImplementedError(
+            f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+        )
+
+
+def _update_matrices_names(
+    file_study: FileStudy,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> None:
+    """
+    Update the matrix file name according to the new operator.
+
+    Args:
+        file_study: the file study
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Raises:
+        NotImplementedError: if the case is not handled
+    """
+
+    if existing_operator == new_operator:
+        return
+
+    parent_folder_node = file_study.tree.get_node(["input", "bindingconstraints"])
+    assert isinstance(parent_folder_node, FolderNode), f"Node type not handled yet: {type(parent_folder_node)}"
+
+    matrix_paths = _infer_input_and_target_paths(
+        parent_folder_node, binding_constraint_id, existing_operator, new_operator
+    )
+
+    # TODO: due to legacy matrices generation, we need to check if the new matrix file already exists
+    #  and if it does, we need to first remove it before renaming the existing matrix file
+
+    if existing_operator != BindingConstraintOperator.BOTH and new_operator != BindingConstraintOperator.BOTH:
+        (matrix_path,) = matrix_paths.matrix_input_paths
+        (new_matrix_path,) = matrix_paths.matrix_target_paths
+        new_matrix_path.unlink(missing_ok=True)
+        matrix_path.rename(new_matrix_path)
+    elif new_operator == BindingConstraintOperator.BOTH:
+        matrix_path_lt, matrix_path_gt = matrix_paths.matrix_target_paths
+        if existing_operator == BindingConstraintOperator.EQUAL:
+            (matrix_path_eq,) = matrix_paths.matrix_input_paths
+            matrix_path_lt.unlink(missing_ok=True)
+            matrix_path_gt.unlink(missing_ok=True)
+            matrix_path_eq.rename(matrix_path_lt)
+            # copy the matrix lt to gt
+            shutil.copy(matrix_path_lt, matrix_path_gt)
+        elif existing_operator == BindingConstraintOperator.LESS:
+            matrix_path_gt.unlink(missing_ok=True)
+            shutil.copy(matrix_path_lt, matrix_path_gt)
+        elif existing_operator == BindingConstraintOperator.GREATER:
```

Comment:
Leaving on an `elif` is a really weird design but I'll think you'll fix it while handling operator equality.